controllers/dogu_requeue_handler.go

```diff
@@ -64,44 +68,48 @@ func NewDoguRequeueHandler(doguInterface doguClient.DoguInterface, recorder reco
 	return &doguRequeueHandler{
 		doguInterface:  doguInterface,
 		nonCacheClient: clientSet,
-		namespace:      namespace,
+		namespace:      operatorConfig.Namespace,
 		recorder:       recorder,
 	}, nil
 }
 
 // Handle takes an error and handles the requeue process for the current dogu operation.
-func (d *doguRequeueHandler) Handle(ctx context.Context, contextMessage string, doguResource *doguv2.Dogu, originalErr error, onRequeue func(dogu *doguv2.Dogu) error) (ctrl.Result, error) {
-	if !shouldRequeue(originalErr) {
+func (d *doguRequeueHandler) Handle(ctx context.Context, doguResource *doguv2.Dogu, err error, reqTime time.Duration) (ctrl.Result, error) {
+	var timeErr error
+	if err == nil && reqTime == 0 {
+		doguResource.Status.RequeueTime = reqTime
+		doguResource.Status.RequeuePhase = getRequeuePhase(err)
+		doguResource, timeErr = d.doguInterface.UpdateStatus(ctx, doguResource, metav1.UpdateOptions{})
+		if timeErr != nil {
+			return ctrl.Result{Requeue: true, RequeueAfter: requeueTime}, fmt.Errorf("failed to set requeue time: %w", timeErr)
+		}
 		return ctrl.Result{}, nil
 	}
+	logger := log.FromContext(ctx)
 
-	requeueTime, timeErr := getRequeueTime(ctx, doguResource, d.doguInterface, originalErr)
-	if timeErr != nil {
-		return ctrl.Result{}, fmt.Errorf("failed to get requeue time: %w", timeErr)
+	if reqTime == 0 {
+		reqTime = requeueTime
 	}
-	if onRequeue != nil {
-		onRequeueErr := onRequeue(doguResource)
-		if onRequeueErr != nil {
-			return ctrl.Result{}, fmt.Errorf("failed to call onRequeue handler: %w", onRequeueErr)
-		}
+
+	doguResource.Status.RequeueTime = reqTime
```

Comment:
This is the same like above

Reply:
I'm unable to see whether this is still relevant.